pkg/controller/s3/bucket/lifecycleConfig.go

```diff
@@ -163,3 +169,14 @@ func (in *LifecycleConfigurationClient) Delete(ctx context.Context, bucket *v1be
 	).Send(ctx)
 	return errors.Wrap(err, lifecycleDeleteFailed)
 }
+
+func sortFilterTags(rules []awss3.LifecycleRule) {
+	if rules == nil {
+		return
+	}
+	for i := range rules {
+		if rules[i].Filter != nil && rules[i].Filter.And != nil {
+			s3.SortS3TagSet(rules[i].Filter.And.Tags)
+		}
+	}
+}
```

Comment:
```suggestion
func sortFilterTags(rules []awss3.LifecycleRule) {
	for i := range rules {
		if rules[i].Filter != nil && rules[i].Filter.And != nil {
			s3.SortS3TagSet(rules[i].Filter.And.Tags)
		}
	}
}
```

That was what I meant. `for` loop won't run at all if `rules == nil` or `len(rules) == 0`. So, we don't need that check there.

Reply:
Ah gotcha, fixed!